### ISSUE TITLE
AddClaimAsync failed with SqlException: Cannot insert the value NULL into column 'Id', table 'dbo.AspNetUserClaims'; column does not allow nulls

### ISSUE THREAD

**Issue by Rick-Anderson** [aspnet/Docs#9518]
Copy from #9433

# Issue description

AddClaimAsync failed  with SqlException: Cannot insert the value NULL into column 'Id', table 'dbo.AspNetUserClaims'; column does not allow nulls

# Software versions
dotnet 2.1.403
NETCore.App 2.1
AspNetCore.App 2.1.5

<details>

According to the instructions in the official document。

[Persist additional claims and tokens from external providers in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/additional-claims?view=aspnetcore-2.1)

when i use code like 
```csharp
await _userManager.AddClaimAsync(user, Principal.FindFirst(ClaimTypes.Gender));
```
i get the error like 
```log
SqlException: Cannot insert the value NULL into column 'Id', table 'guandian.dbo.AspNetUserClaims'; column does not allow nulls. INSERT fails.

DbUpdateException: An error occurred while updating the entries. See the inner exception for details.

```

</details>

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 64faf33b-2511-5e8f-c1b0-94f32dc07fa4
* Version Independent ID: 2697abc8-7727-2eba-74a5-fbbc8a7c80ff
* Content: [Persist additional claims and tokens from external providers in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/additional-claims?view=aspnetcore-2.1)
* Content Source: [aspnetcore/security/authentication/social/additional-claims.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/security/authentication/social/additional-claims.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
@guardrex 	please review. cc @niltor

**Comment by cncgate** [comment]
I run into same error and was able to identify reason. 
The issue happened, because I created initial ModelSnapshot against **SQLite** database. Later switching to SQL Server, it was crashing with the same error and I deleted ModelSnapshot and re-created again - against current **SQL Server**.
The difference in shapshot file was in lines related to keys:
`.HasAnnotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn);`
for tables IdentityRoleClaim and IdentityUserClaim.
So running update-database with fresh ModelSnapshot solved the issue for me.
